extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/JdbcCacheGenerator.java

```diff
@@ -101,23 +109,41 @@
         }
     );
     final String newVersion;
-    if (lastDBUpdate != null) {
-      newVersion = lastDBUpdate.toString();
-    } else {
-      newVersion = StringUtils.format("%d", dbQueryStart);
-    }
-    final CacheScheduler.VersionedCache versionedCache = scheduler.createVersionedCache(entryId, newVersion);
+    CacheScheduler.VersionedCache versionedCache = null;
     try {
-      final Map<String, String> cache = versionedCache.getCache();
-      for (Pair<String, String> pair : pairs) {
-        cache.put(pair.lhs, pair.rhs);
+      if (doIncrementalLoad) {
+        newVersion = StringUtils.format("%d", lastDBUpdate);
+        ConcurrentMap<String, String> newCachedEntries = new ConcurrentHashMap<>();
+        LOG.info(" found %s new incremental entries for %s", pairs.size(), entryId);
+        for (Pair<String, String> pair : pairs) {
+          newCachedEntries.put(pair.lhs, pair.rhs);
+        }
+        versionedCache = entryId.createFromExisitngCache(entryId, newVersion, newCachedEntries);
+        return versionedCache;
+      } else {
+        LOG.info("Not doing incremental load since lastDBUpdate: %s, namespace.getTsColumn() %s, lastVersion %s," +
+              " lastCheck %s", lastDBUpdate, namespace.getTsColumn(), lastVersion, lastCheck);
```

Comment:
Not correct formatting of multiline arguments. All args should be chopped down. See an example in `buildLookupQuery()`.

Reply:
done